core/templates/dev/head/pages/topic-viewer-page/topic-viewer-page.controller.ts

```diff
@@ -65,14 +65,14 @@ angular.module('oppia').directive('topicViewerPage', [
 
             $rootScope.loadingMessage = 'Loading';
             TopicViewerBackendApiService.fetchTopicData(ctrl.topicName).then(
-              function(topicDataDict) {
-                ctrl.topicId = topicDataDict.topic_id;
-                ctrl.canonicalStoriesList = topicDataDict.canonical_story_dicts;
-                ctrl.degreesOfMastery = topicDataDict.degrees_of_mastery;
-                ctrl.skillDescriptions = topicDataDict.skill_descriptions;
-                ctrl.subtopics = topicDataDict.subtopics;
+              function(topicDataObject) {
+                ctrl.topicId = topicDataObject.getTopicId();
+                ctrl.canonicalStoriesList =
+                  topicDataObject.getCanonicalStories();
+                ctrl.degreesOfMastery = topicDataObject.getDegreesOfMastery();
+                ctrl.skillDescriptions = topicDataObject.getSkillDescription();
+                ctrl.subtopics = topicDataObject.getSubtopics();
```

Comment:
I don't see any usage of additionalStory, uncategorizedSkillIds, skillDescriptions here, is this `TopicViewerBackendApiService` being used anywhere else? If not then I'm not sure why we are sending the data through API if it's not used anywhere? @aks681, Can you explain?

Reply:
@aks681, Can you please reply to this?

Reply:
No, it's not used anywhere else.
Yeah, we don't use additionalStories now anywhere actually, but I think it's fine to leave that in since it will always be an empty list.
Not sure of uncategorizedSkillIds, but I thought we used skill descriptions in the practice sessions tab...

Reply:
Hey @aks681 , can we get `node_count` value from any other class or can we assign it vague value? as I'm short of only 1 field here.
![image](https://user-images.githubusercontent.com/32016324/74030890-2a50f880-49d6-11ea-9142-51dc4b5d0762.png)

Reply:
Oh, I was pretty sure the objects in canonical_story_dicts list had that field, can you check?

Reply:
I did `console.log(response.data)` in `service.ts` file & didn't find it. My flow was right?
![image](https://user-images.githubusercontent.com/32016324/74055746-9b100900-4a06-11ea-8b56-facf241a104b.png)

Reply:
Hi, following our discussions I made changes in backend

Reply:
Following our discussions I made changes in backend.